File: DemoServer/AdditionalLanguages/go/Basics/theDocumentStore/TheDocumentStore.go
package main

//region Usings
import "github.com/ravendb/ravendb-go-client"
//endregion

func main() {
    createDocumentStore()
}

var globalDocumentStore *ravendb.DocumentStore

//region Demo
//region Step_1
var globalDocumentStore *ravendb.DocumentStore
//endregion

//region Step_2
func createDocumentStore() (*ravendb.DocumentStore, error) {
//endregion    
  
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }

    //region Step_3 
    serverURL := "http://localhost:8080"
    databaseName := "YourDatabaseName"
    
    urls := []string {serverURL}
    store := ravendb.NewDocumentStore(urls, databaseName)
    //endregion

    //region Step_4
    store.Certificate = ... 
    store.TrustStore = ...
    //endregion
    
    //region Step_5
    conventions := ravendb.NewDocumentConventions() 
    // Modify conventions as needed
    store.SetConventions(conventions) 
    //endregion

    //region Step_6
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    //endregion

    //region Step_7
    globalDocumentStore = store
    //endregion
    
    return globalDocumentStore, nil
}
//endregion

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/basics/theDocumentStore/TheDocumentStore.java
package net.ravendb.demo.basics.theDocumentStore;
//region Usings
import net.ravendb.client.documents.IDocumentStore;
import net.ravendb.client.documents.DocumentStore;
package com.journaldev.singleton;
//endregion

//region Demo
//region Step_1
public class DocumentStoreHolder {
//endregion
    //region Step_2
    private static IDocumentStore store;
    //endregion

    //region Step_3
    private IDocumentStore createDocumentStore()
    //endregion
        //region Step_4
        string serverURL = "http://localhost:8080";
        string databaseName = "YourDatabaseName";

        IDocumentStore documentStore = new DocumentStore
        {
            Urls = new[] { serverURL },
            Database = databaseName
        };
        //endregion

        //region Step_5
        documentStore.initialize();
        //endregion
        return documentStore;
    }

    //region Step_6
    public static IDocumentStore getStore() {
        if(store == null) {
            store = new createDocumentStore();
        }

        return store;
    }
    //endregion
}
//endregion
